### PR TITLE
#166265825 Setup Eslint

### PR DESCRIPTION
**Pull request by Hadeneekeh**
#### What does this PR do?
* Have eslint configuration

#### Description of Task to be completed?
* Install eslint, eslint-config-airbnb-base and eslint-plugin-import
* Add eslint configuration file
* Add the eslint style guide requested by the Product Owner

#### How should this be manually tested?
* clone the repo git clone `https://github.com/emp-daisy/HelloBooksLib.git`

* on your terminal run `cd HelloBooksLib`

* run `npm install`

#### Any background context you want to provide?
NA

#### What are the relevant pivotal tracker stories?
[#166265825](https://www.pivotaltracker.com/n/projects/2351121/stories/166265825 "Setup Eslint")

#### Screenshots (if appropriate)
N/A

### CODE REVIEW

**Review by encodedBicoding — COMMENTED** [review]
why did you substitute `jest` for `eslint-plugin-import`. I though @jude used the jest on his test?

File: .eslintrc.json [hunk 1]
```diff
@@ -0,0 +1,34 @@
+{
+  "root": true,
+  "extends": "airbnb-base",
```
Comment (OvieMudi) [inline .eslintrc.json]: I think ESLint also need to extend Prettier to avoid conflict.
I'm not quite sure about this, so guys please come in here.

Comment (Hadeneekeh) [reply .eslintrc.json]: I will read about that, thank you.

**Comment by Hadeneekeh** [comment]
> why did you substitute `jest` for `eslint-plugin-import`. I though @jude used the jest on his test?

I just did pull rebase to resolve conflict, I guess that must have been done on someone's PR which have been merged to the develop branch.

File: .eslintrc.json (hunk 1 above)
Comment (OvieMudi) [reply .eslintrc.json]: Oh sorry for not being clear, I meant Linting conflict. Not merge conflict.

File: package.json [hunk 2]
```diff
@@ -37,11 +37,10 @@
     "@babel/node": "^7.4.5",
     "@babel/preset-env": "^7.4.5",
     "@babel/register": "^7.4.4",
-    "chai": "^4.2.0",
-    "chai-http": "^4.3.0",
-    "dotenv": "^8.0.0",
+    "eslint": "^5.16.0",
```
Comment (emp-daisy) [inline package.json:44]: Confirm you didn't overwrite some packages required

Comment (Hadeneekeh) [reply package.json:44]: I did `git pull --rebase origin develop`, then there was a message that there's conflict that was not resolved in package-lock.json file so I accepted all incoming changes and did `git rebase --continue`. Thereafter, I did `git push origin my current branch` but I got an error message that my current branch is behind its remote counterpart so I did `git pull`. By recounting the process now, I realised that my last step must have overwritten the package.

I will make corrections.

**Review by PelumiAlesh — DISMISSED** [review]

**Review by OvieMudi — DISMISSED** [review]

**Review by chuxmykel — COMMENTED** [review]
Can you confirm why Travis is failing?

**Review by encodedBicoding — COMMENTED** [review]
ANy thoughts on why build is failing?

File: package.json (hunk 2 above)
Comment (emp-daisy) [reply package.json:44]: Next time you have an issue with package-lock.json. Just do ‘npm install’ and it will correct it

Comment (Hadeneekeh) [reply package.json:44]: Thank you, I just realised that now while troubleshooting the travis build error.

Comment (emp-daisy) [reply package.json:44]: So just do that to regenerate the lock file. That should resolve your Travis fail

Comment (Hadeneekeh) [reply package.json:44]: Done. Thank you

**Review by emp-daisy — APPROVED** [review]

**Review by OvieMudi — APPROVED** [review]

**Review by chuxmykel — APPROVED** [review]
LGTM

**Pull request merged**